homeassistant/components/decora_wifi/__init__.py

```diff
@@ -1 +1,104 @@
 """The decora_wifi component."""
+
+import logging
+
+from decora_wifi import DecoraWiFiSession
+from decora_wifi.models.iot_switch import IotSwitch
+from decora_wifi.models.permission import Permission
+from decora_wifi.models.person import Person
+from decora_wifi.models.residence import Residence
+from decora_wifi.models.residential_account import ResidentialAccount
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+
+from .config_flow import BASE_SCHEMA
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = [Platform.LIGHT]
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: BASE_SCHEMA},
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    session = DecoraWiFiSession()
+    user = await hass.async_add_executor_job(lambda: session.login(username, password))
+    if not user:
+        raise ConfigEntryError("could not authenticate")
+
+    hass.data[DOMAIN][entry.entry_id] = session
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    def logout(event: Event) -> None:
+        if session is not None:
+            try:
+                Person.logout(session)
+            except ValueError as err:
+                _LOGGER.error("Failed to log out of myLeviton Service: %s", err)
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, logout)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, logout)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    hass.data.pop(DOMAIN)
+    return unload_ok
+
+
+class DecoraWifiAsyncClient:
+    """A thin wrapper to make async calls more readable."""
+
+    def __init__(self, session: DecoraWiFiSession, hass: HomeAssistant) -> None:
+        """DecoraWifiAsyncClient."""
+        self.session = session
+        self.hass = hass
+
+    async def get_permissions(self):
```

Comment:
Can we type these?

Reply:
https://github.com/home-assistant/core/pull/104573/commits/0294a93e9865aedb7cbce99526fcddd9e3ff510e#diff-2776b1a16ddfe5f22bf597804e9c8a179a4d7b9313af55a731fbdecc2ae030f7R85